core/src/components/datetime/datetime-util.ts

```diff
@@ -324,6 +324,21 @@ export const updateDate = (existingData: DatetimeData, newData: any): boolean =>
         (existingData as any)[key] = newData[key].value;
       }
       return true;
+    } else if (newData.ampm) {
+      if (!newData.hour) {
+        // Even though in the picker column hour values are between 1 and 12, the hour value is actually normalized
+        // to [0, 23] interval. Because of this when changing between AM and PM we have to update the hour so it points
+        // to the correct HH hour
+        newData.hour = {
+          value: newData.hour
```

Comment:
If `newData.hour` is guaranteed to be undefined/null, why do we need this additional check?

Reply:
You're right, it's unnecessary. I'll remove it.